packages/effect/src/internal/rateLimiter.ts

```diff
@@ -0,0 +1,78 @@
+/**
+ * This is a direct port of `RateLimiter` from Rezilience
+ * https://github.com/svroonland/rezilience/blob/master/rezilience/shared/src/main/scala/nl/vroste/rezilience/RateLimiter.scala
+ */
+
+import * as Chunk from "../Chunk.js"
+import * as Deferred from "../Deferred.js"
+import type { DurationInput } from "../Duration.js"
+import * as Effect from "../Effect.js"
+import { pipe } from "../Function.js"
+import * as Queue from "../Queue.js"
+import * as Ref from "../Ref.js"
+import * as Stream from "../Stream.js"
+import { nextPow2 } from "./nextPow2.js"
+
+/** @internal */
+export const make = (limit: number, window: DurationInput) => {
+  return Effect.gen(function*($) {
+    const q = yield* $(Queue.bounded<[Ref.Ref<boolean>, Effect.Effect<never, never, void>]>(nextPow2(limit)))
+
+    yield* $(
+      pipe(
+        Stream.fromQueue(q, { maxChunkSize: 1 }),
+        Stream.filterEffect(([interrupted]) => {
+          return pipe(
+            Ref.get(interrupted),
+            Effect.map((b) => !b)
+          )
+        }),
+        Stream.throttle({
+          strategy: "shape",
+          duration: window,
+          cost: Chunk.size,
+          units: limit
+        }),
+        Stream.mapEffect(([_interrupted, eff]) => eff, { concurrency: "unbounded", unordered: true }),
+        Stream.runDrain,
+        Effect.forkScoped
+      )
+    )
+
+    const apply = <R, E, A>(task: Effect.Effect<R, E, A>) =>
+      Effect.gen(function*($) {
+        const start = yield* $(Deferred.make<never, void>())
```

Comment:
I think we're seeing all the type errors in CI because we've re-ordered the type parameters of many of the data types within Effect (cc @mikearnaldi @tim-smart). It's a massive breaking change that we've actually developed the Effect [`@effect/codemod`](https://github.com/Effect-TS/codemod) library to fix these issues automatically.

You can either use the codemod lib to fix these changes, or you can fix the types manually, and since we re-ordered the types you should now be able to do:

```ts
Effect<void>
```

```ts
Deferred.make<void>()
```

Reply:
Done.